src/provider/index.ts

```diff
@@ -128,7 +128,7 @@ export class Main {
     private async findIntent(payload: FindIntentPayload): Promise<AppIntent> {
         let apps: Application[];
         if (payload.intent) {
-            apps = await this._model.getApplicationsForIntent(payload.intent);
+            apps = await this._model.getApplicationsForIntent(payload.intent, payload.context && parseContext(payload.context).type);
```

Comment:
that's essentially getAppIntentsByContext() with it's dedicated API `APIFromClientTopic.FIND_INTENTS_BY_CONTEXT`. should we replicate it (or redirect to it from) here?

Reply:
Hmm, actually, digging a bit deeper I'm more confused. Going by the interface, `payload.intent` is required.

Hey @pjbroadbent - the workaround here where `findIntent` can end up returning all apps. Can we get rid of it? AFAICT, it's unused and undocumented.

And I know this is scope creeping now, but the value we're giving the the `displayName` for the intent. We should be smarter about this, right?

Reply:
@bryangaleOF @pjbroadbent 
So, in our latest conversation we concluded, the workaround Bryan mentions here is still needed as that's currently the only way to get all apps for an intent which is undocumented because it still requires the workgroup agreement but needed in the near future, and is used by the demo apps. So I am reverting this back to include the workaround. Are we all happy with this?

Reply:
Yeah, fine with the workaround staying since something's using it. Though then there's still the question of how to handle the "context specified, but intent not" case

Reply:
I think the only case we use "no intent" is when we want to get all apps (as this is the main reason for having this temporary workaround) maybe we shouldn't worry about context at all in that case?

Reply:
Yeah, I'm good with that

Reply:
Yeah, I'm good with that

Reply:
👍

Reply:
Will resolve with once we have a story in for a better displayName. Shout if you want me to write that up

Reply:
👍